### PR TITLE
UIMARCAUTH-75 Searching MARC authority records: Split up Exclude See from filters into two filters

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://issues.folio.org/browse/UIMARCAUTH-75
## Approach

Сurrent implementation of the `References` filter relies on the `isExcludedSeeFromLimiter` state value in the `AuthoritiesSearchContext`and is not related to the state of the filters. This approach looks inconsistent with other filters.

Steps to split up Exclude See from filters into two filters:

- Add filter component (`ReferencesFilter`) to handle two options: `Exclude see from` and `Exclude see from also`;
- Add handler to the new filter in `filterConfig`;
- Update `useAuthorities`, `useBrowseAuthorities` hooks and their utils to support the filter;
- Remove redundant `isExcludedSeeFromLimiter` state from the code;

## Screenshots

msedge_tpHmqJlgoU

### CODE REVIEW

**Review by AndreiShumski2 — APPROVED** [review]

**Review by BogdanDenis — APPROVED** [review]
Good work, thank you!

File: src/queries/utils/buildQuery.test.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import buildQuery from './buildQuery';
-import { searchableIndexesValues } from '../../constants';
+import { FILTERS, REFERENCES_VALUES_MAP, searchableIndexesValues } from '../../constants';
```
Comment (mariiaschv) [inline src/queries/utils/buildQuery.test.js:2]:
```suggestion
import {
  FILTERS,
  REFERENCES_VALUES_MAP,
  searchableIndexesValues,
} from '../../constants';
```

File: src/constants/filterConfig/filterConfig.test.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import { filterConfig } from './filterConfig';
+import { REFERENCES_VALUES_MAP } from '../references';
+import { filterConfig, FILTERS } from './filterConfig';
```
Comment (mariiaschv) [inline src/constants/filterConfig/filterConfig.test.js:2]:
```suggestion
import {
  filterConfig,
  FILTERS,
} from './filterConfig';
```

File: src/components/SearchFilters/ReferencesFilter/ReferencesFilter.test.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import user from '@testing-library/user-event';
+import { act, render, screen } from '@testing-library/react';
```
Comment (mariiaschv) [inline src/components/SearchFilters/ReferencesFilter/ReferencesFilter.test.js:2]:
```suggestion
import {
  act,
  render,
  screen,
} from '@testing-library/react';
```

**Review by mariiaschv — APPROVED** [review]
LGTM!
I'd only suggest to place destructured values on separate lines when importing

**Pull request merged**